docs/machine-learning/basic-concepts-model-training-in-mldotnet.MD

```diff
@@ -0,0 +1,169 @@
+- [*Transformer*](#transformer), represented as `ITransformer` interface.
+  - In one sentence, a transformer is a component that takes data, does some work on it, and returns new 'transformed' data.
+  - For example, you can think of a machine learning model as a transformer that takes features and returns predictions.
+  - Another example, 'text tokenizer' would take a single text column and output a vector column with individual 'words' extracted out of the texts.
+- [*Data reader*](#data-reader), represented as an `IDataReader<T>` interface.
+  - The data reader is ML.NET component to 'create' data: it takes an instance of `T` and returns data out of it. 
+  - For example, a *TextLoader* is an `IDataReader<FileSource>`: it takes the file source and produces data. 
+- [*Estimator*](#estimator), represented as an `IEstimator<T>` interface.
+  - This is an object that learns from data. The result of the learning is a *transformer*.
+  - You can think of a machine learning *algorithm* as an estimator that learns on data and produces a machine learning *model* (which is a transformer).
+- [*Prediction function*](#prediction-function), represented as a `PredictionFunction<TSrc, TDst>` class.
+  - The prediction function can be seen as a machine that applies a transformer to one 'row', such as at prediction time.
+
+## Data
+
+In ML.NET, data is very similar to a SQL view: it's a lazily-evaluated, cursorable, heterogenous, schematized dataset.
+
+- It has *Schema* (an instance of an `ISchema` interface), that contains the information about the data view's columns.
+  - Each column has a *Name*, a *Type*, and an arbitrary set of *metadata* associated with it.
+  - It is important to note that one of the types is the `vector<T, N>` type, which means that the column's values are *vectors of items of type T, with the size of N*. This is a recommended way to represent multi-dimensional data associated with every row, like pixels in an image, or tokens in a text.
+  - The column's *metadata* contains information like 'slot names' of a vector column and suchlike. The metadata itself is actually represented as another one-row *data*, that is unique to each column.
+- The data view is a source of *cursors*. Think SQL cursors: a cursor is an object that iterates through the data, one row at a time, and presents the available data.
+  - Naturally, data can have as many active cursors over it as needed: since data itself is immutable, cursors are truly independent.
+  - Note that cursors typically access only a subset of columns: for efficiency, we do not compute the values of columns that are not 'needed' by the cursor.
+
+## Transformer
+
+A transformer is a component that takes data, does some work on it, and return new 'transformed' data.
+
+Here's the interface of `ITransformer`:
+
+```csharp
+public interface ITransformer
+{
+    IDataView Transform(IDataView input);
+    ISchema GetOutputSchema(ISchema inputSchema);
```

Comment:
Is ISchema still used? I thought ITransformer now provided Schema directly.

/cc @Zruty0

Reply:
I  don't think the interfaces discussion brings value at this time. Removing.